extras/rya.pcj.fluo/pcj.fluo.app/src/main/java/org/apache/rya/indexing/pcj/fluo/app/query/MetadataCacheSupplier.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.rya.indexing.pcj.fluo.app.query;
+
+import java.util.concurrent.locks.ReentrantLock;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MetadataCacheSupplier {
+
+    private static final Logger LOG = LoggerFactory.getLogger(MetadataCacheSupplier.class);
+    private static FluoQueryMetadataCache CACHE;
+    private static boolean initialized = false;
+    private static final int DEFAULT_CAPACITY = 10000;
+    private static final int DEFAULT_CONCURRENCY = 8;
+    private static final ReentrantLock lock = new ReentrantLock();
+
+    /**
+     * Returns an existing cache with the specified instance name, or creates a cache. The created cache will have the
```

Comment:
I don't see an instance name here. I think this class only supports a single cache.

Reply:
It does.  It enforces "singleton like" behavior in that it only creates the instance if it doesn't already exist.  It allows for the initial cache construciton to be configurable.